certifikit/src/commonMain/kotlin/app/cash/certifikit/text/pem.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.certifikit.text
+
+import app.cash.certifikit.KeyPair
+import app.cash.certifikit.PrivateKey
+import app.cash.certifikit.X509Certificate
+import okio.ByteString
+
+/**
+ * Decodes a multiline string that contains a [certificate][certificatePem] which is
+ * [PEM-encoded][rfc_7468]. A typical input string looks like this:
+ *
+ * ```
+ * -----BEGIN CERTIFICATE-----
+ * MIIBYTCCAQegAwIBAgIBKjAKBggqhkjOPQQDAjApMRQwEgYDVQQLEwtlbmdpbmVl
+ * cmluZzERMA8GA1UEAxMIY2FzaC5hcHAwHhcNNzAwMTAxMDAwMDA1WhcNNzAwMTAx
+ * MDAwMDEwWjApMRQwEgYDVQQLEwtlbmdpbmVlcmluZzERMA8GA1UEAxMIY2FzaC5h
+ * cHAwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAASda8ChkQXxGELnrV/oBnIAx3dD
+ * ocUOJfdz4pOJTP6dVQB9U3UBiW5uSX/MoOD0LL5zG3bVyL3Y6pDwKuYvfLNhoyAw
+ * HjAcBgNVHREBAf8EEjAQhwQBAQEBgghjYXNoLmFwcDAKBggqhkjOPQQDAgNIADBF
+ * AiAyHHg1N6YDDQiY920+cnI5XSZwEGhAtb9PYWO8bLmkcQIhAI2CfEZf3V/obmdT
+ * yyaoEufLKVXhrTQhRfodTeigi4RX
+ * -----END CERTIFICATE-----
+ * ```
+ */
+expect fun String.decodeCertificatePem(): X509Certificate
+
+/**
+ * Returns the certificate encoded in [PEM format][rfc_7468].
+ *
+ * [rfc_7468]: https://tools.ietf.org/html/rfc7468
+ */
+expect fun X509Certificate.certificatePem(): String
```

Comment:
I don’t know how I feel about these extensions, especially on types we don’t control. Seems like it could collide with a platform method with similar but incompatible behavior?

Reply:
True, part of the problem is 

![image](https://user-images.githubusercontent.com/231923/147981880-1eaef207-7548-4d89-958b-4ac007c4eb47.png)

I'm trying to upskill on KMP, but also exploring what the JS implementations looks like.  

For BigInteger, after looking at BigInt, BigUint64Array, I'm now leaning towards ByteArray. Since actually most of the conversions in javascript are to this, and it means the implementations can be kt files instead of js() blocks.

Same for X509Certificate - probably https://nodejs.org/api/crypto.html#class-x509certificate? But the expect modality may not line up.

Reply:
This has been there for a while

https://github.com/cashapp/certifikit/commit/1a860c886913770ea8057059019fd68d7e740c63#diff-0935cc35c6ffb3441fa251667a8fccdbeca4164300b4f295e54ad6c35828c4c0

I agree with your concerns, but it's actually some of the highest value functionality we can offer as a library.